cmd/kubeadm/app/phases/addons/dns/dns.go

```diff
@@ -297,6 +298,88 @@ func createDNSService(dnsService *v1.Service, serviceBytes []byte, client client
 	return nil
 }
 
+func createCoreDNSConfigMap(client clientset.Interface, cm *v1.ConfigMap) error {
+	// Get the CoreDNS ConfigMap. If not present, Create a new default ConfigMap.
+	coreDNSConfigMap, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Get(kubeadmconstants.CoreDNSConfigMap, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil
+		}
+		if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Create(cm); err != nil {
+			if !apierrors.IsAlreadyExists(err) {
+				return errors.Wrap(err, "unable to create configmap")
+			}
+		}
+		return nil
+	}
+	// Get the Corefile and installed CoreDNS version.
+	corefile, currentInstalledCoreDNSVersion, err := GetCoreDNSInfo(client, coreDNSConfigMap)
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch CoreDNS current installed version and ConfigMap.")
+	}
+
+	// If the current Configuration present is the default (unmodified) version, just update it to the latest default.
+	if migration.Default("", corefile) {
+		if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Update(cm); err != nil {
+			return errors.Wrap(err, "unable to update the CoreDNS ConfigMap")
+		}
+	} else {
+		// Since the current Configuration present is the not the default version, try and migrate the Config.
+		updatedCorefile, err := migration.Migrate(currentInstalledCoreDNSVersion, kubeadmconstants.CoreDNSVersion, corefile, false)
+		if err != nil {
+			return errors.Wrap(err, "unable to migrate CoreDNS ConfigMap")
+		}
+
+		if _, err := client.CoreV1().ConfigMaps(cm.ObjectMeta.Namespace).Update(&v1.ConfigMap{
```

Comment:
what does this do to currently-running CoreDNS pods? do they try to reload the updated/migrated config? if so, will that work?

if the upgrade process stops, does this leave the CoreDNS configmap populated with data that breaks the 1.14-level of CoreDNS?

I expected to leave any 1.14-level configmap alone, so the current CoreDNS pods would continue working, create a 1.15-level configmap and point the 1.15 deployment of CoreDNS at that config